Source/settings/Settings.js

```diff
@@ -141,3 +140,32 @@ export function setDocumentSettingForKey(document, key, value) {
   }
   __command.setValue_forKey_onDocument(stringifiedValue, key, documentData)
 }
+
+export function sessionVariable(key) {
+  const threadDic = NSThread.mainThread().threadDictionary()
+
+  const value = threadDic.objectForKey(
+    `${SUITE_PREFIX}${getPluginIdentifier()}.${key}`
+  )
+
+  if (typeof value === 'undefined' || value == 'undefined' || value === null) {
+    return undefined
+  }
+  return JSON.parse(value)
+}
+
+export function setSessionVariable(key, value) {
+  const threadDic = NSThread.mainThread().threadDictionary()
+
+  const stringifiedValue = JSON.stringify(value, (k, v) => util.toJSObject(v))
+  if (!stringifiedValue) {
+    threadDic.removeObjectForKey(
```

Comment:
Maybe I'm reading this wrong, but… if a plugin tries to store something in the session variable that can't be stringified, we erase the existing contents of that variable? Isn't this a dangerous thing to do? If I already have something there and there's a bug in my code that tries to store something that can't be stringified, I'll lose the existing value… 🤔

Reply:
if it can't be stringified, it will crash before trying to set anything so I don't think it's an issue.

But actually most of the times _I_ use the thread dictionary, it's to store a native object that I don't want to stringify 🤔